airflow/sensors/base_sensor_operator.py

```diff
@@ -65,6 +89,11 @@ def poke(self, context):
 
     def execute(self, context):
         started_at = timezone.utcnow()
+        if self.reschedule:
+            # If reschedule, use first start date of current try
+            task_reschedules = TaskReschedule.find_for_task_instance(context['ti'])
+            if task_reschedules:
+                started_at = task_reschedules[0].start_date
         while not self.poke(context):
             if (timezone.utcnow() - started_at).total_seconds() > self.timeout:
```

Comment:
Normally that should not be the case because `started_at` is always in the past. Of course clocks are never in sync (except at Google) so it may happen. But also in such a case `datetime.timedelta.total_seconds()` doesn't throw an exception but returns a negative number which is still smaller than the configured timeout (if the user configured a positive timeout, lol)